src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
Just returning null here provides the correct behavior.  I'm not certain why we would try to add more complexity.

1. `dotnet restore` has already failed, telling the user the dependency doesn't exist.
2. If I try running `dotnet build` on this failed-to-restore project it prints out:

```
C:\cli\TestAssets\NonRestoredTestProjects\TestProjectWithUnresolvedPlatformDependency\project.json(7,103): error NU1001: The dependency ThisIsNotARealDependencyAndIfSomeoneGoesAndAddsAProjectWithThisNameIWillFindThemAndPunishThem >= 1.0.0 could not be resolved.
```